sdk/src/main/scala/com/horizen/account/receipt/EthereumReceipt.scala

```diff
@@ -101,9 +116,9 @@ object EthereumReceiptSerializer extends SparkzSerializer[EthereumReceipt]{
     writer.putInt(gasUsedBytes.length)
     writer.putBytes(gasUsedBytes)
 
-    //optional field
-    writer.putInt(receipt.contractAddress.length)
-    writer.putBytes(receipt.contractAddress)
+    // optional field
+    writer.putInt(receipt.contractAddress.get.length)
+    writer.putBytes(receipt.contractAddress.get)
```

Comment:
Making `contractAddress` an `Option[]` is good, but just calling `get()` everywhere is not using it right. This will throw exceptions whenever it is `None`. Instead we need to explicitly handle the `None` case.

You could do something like this:
```suggestion
    val addr = receipt.contractAddress.getOrElse(Array.empty)
    writer.putInt(addr.length)
    writer.putBytes(addr)
```